Move yast2_i test modules to YaST job group

b10n1k opened the pull request:
Added svirt-hyperv schedule for yast2_ncurses_gnome. Need to setup                                                                                                                                                                           
the job group accordingly.

- Related ticket: https://progress.opensuse.org/issues/66856
- Functional group yaml: https://gitlab.suse.de/qsf-u/qa-sle-functional-userspace/-/merge_requests/88
- Yast group yaml: https://gitlab.suse.de/qsf-y/qa-sle-functional-y/-/merge_requests/205
- Verification run: [summary](https://openqa.suse.de/tests/overview?build=b10n1k%2Fos-autoinst-distri-opensuse%23yast_i&distri=sle&version=15-SP2)

jknphy commented on schedule/functional/allpatterns.yaml:
```diff
@@ -47,7 +47,6 @@ schedule:
     - console/glibc_sanity
     - console/zypper_in
     - console/zypper_log
-    - console/yast2_i
```
This module is about software installation so I think we should cover this part this part deleted here as well: https://openqa.suse.de/tests/4308925# adding the module to this test suite, so we could have more chances to catch a bug, wdyt?

b10n1k replied:
i was checking this suite as well but for the FullMedia instead of Online. But i let it out of this PR first because of simplicity and second because according to the description indicates testing on _enabling all modules and selecting all patterns._ So i do not think that the yast2_i applies here but ofc we can do it. However the question for me is Online or Full media(as we have some coverage already for Online)

jknphy replied:
In functional groups there was only one job and for that medium is still work in progress for both teams to add more tests there, so I would say that for Online at least we should cover that.

jknphy commented:
I noticed in your VR you are missing s390x and hyperv. Please provide for those as well.

jknphy commented:
> > I noticed in your VR you are missing s390x and hyperv. Please provide for those as well.
> 
> I added the s390x VR. Also module added for allmodules+allpatterns. But there was an issue with hyperv. What do you say to leave it outside for now as it may need some extra work?

Please, check that is failing in the same way in functional group and added it anyway in that case (as we are not extending cover in the scope of this ticket). If fails for other reason then we will need to investigate.

b10n1k commented:
> > > I noticed in your VR you are missing s390x and hyperv. Please provide for those as well.
> > 
> > 
> > I added the s390x VR. Also module added for allmodules+allpatterns. But there was an issue with hyperv. What do you say to leave it outside for now as it may need some extra work?
> 
> Please, check that is failing in the same way in functional group and added it anyway in that case (as we are not extending cover in the scope of this ticket). If fails for other reason then we will need to investigate.

the reason is different and not related to the module

jknphy commented:
> > > > I noticed in your VR you are missing s390x and hyperv. Please provide for those as well.
> > > 
> > > 
> > > I added the s390x VR. Also module added for allmodules+allpatterns. But there was an issue with hyperv. What do you say to leave it outside for now as it may need some extra work?
> > 
> > 
> > Please, check that is failing in the same way in functional group and added it anyway in that case (as we are not extending cover in the scope of this ticket). If fails for other reason then we will need to investigate.
> 
> the reason is different and not related to the module

then we can use some other virtualization, as we discussed that we want it at least one to achieve similar test coverage.

b10n1k commented:
> suggested to add another virtualization job.

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/c685e0ae0429a898f00ea03d755add9ea9601186/schedule/yast/yast2_ncurses_textmode%40svirt-xen-hvm.yaml

jknphy commented on schedule/yast/yast2_ncurses_textmode@svirt-xen-hvm.yaml:
```diff
@@ -0,0 +1,13 @@
+---
+name:           yast2_ncurses_textmode@svirt-hyperv
```
if more than one, let's create a folder `schedule/yast/yast2_ncurses/` wdyt?

b10n1k replied:
Can be

b10n1k replied:
Done. Updated the jobGroup as well

jknphy commented on schedule/yast/yast2_ncurses_gnome.yaml:
```diff
@@ -1,50 +1,51 @@
+  - boot/boot_to_desktop
+  - console/prepare_test_data
+  - console/consoletest_setup
+  - console/yast2_i
```
could you please put this module inside the conditions repeating it for the 4 archs? As I could check in other PR the order matters and having conditions for all architecture it is more clear to use the same order and repeat the modules. Also it is easy to understand why yast2_i is outside {{yast2_modules}}. In further work we can optimize, but for now I'd like that we follow this suggestion. You don't need extra verification for that, just put it in the original order.

b10n1k replied:
Sorry, but nothing indicates that this module plays any role in the order.

jknphy replied:
not this one, but it has been proved that is confusing for this tests to have some common part when you are still using all the the arch to do the conditional, basically it is an invitation for the next one to fail and spend time searching for common denominator in this kind of tests. But still if you want to keep it it is fine for me.

b10n1k replied:
In my point of you, if this causes any problem, it would mean a bug. Fixing the order to make it work, it would be not ideal as it would hide the problem. Maybe i dont understand something. But i really believe that is more than ok even as it is

jknphy commented on tests/console/yast2_i.pm:
```diff
@@ -20,7 +20,7 @@
 # - Disable "recommended packages" and accept install of test package
 # - Remove test package
-# Maintainer: Martin Kravec <mkravec@suse.com>
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
```
completely unnecessary but up to you... we already talked offline about it, and if you are not changing a lot this module keep the original writer (even if the mantainer field is useless) is fine.

b10n1k replied:
IMO it is essential. Martin is not part of the team and we should take full responsibility.

jknphy replied:
we should not be touching headers/licenses of files that we don't modify extensively, the current maintainer might still be the best contact person and we might not know very well the code in this module. if someone or some other team reach us  with this we might still consider not schedule this module at all, or create a complete different module.  There is a lot of shared code in the repo, for that reason it doesn't make sense at all, and we should narrow our responsibility to what we know very well. Still if you want to do it, do it, no problem for me, it is a YaST module after all.

b10n1k replied:
Obviously i am not aware of many of the above. but this does not change that this is our responsibility either we know the module well or not.

rwx788 commented:
https://gitlab.suse.de/qsf-u/qa-sle-functional-userspace/-/merge_requests/88 has conflict, let's resolve and proceed further.

b10n1k commented:
> https://gitlab.suse.de/qsf-u/qa-sle-functional-userspace/-/merge_requests/88 has conflict, let's resolve and proceed further.

That was expected. and it resolved

rwx788 commented:
Please, restart CI on the job group PR to see if we have missed anything.
